### ISSUE TITLE
ATK usage

### ISSUE THREAD

**Issue by giuseppe1993** [afcowie/java-gnome#21]
I'm working on a project to create a bridge between Java and Gnome accessibility. The solution already exists is Java Accessibility Wrapper but has a lot of problems. I would like to use Java-Gnome as ATK binding in a Java class and add other classes to complete this bridge. I have already tried to look into Java-Gnome and I don't understand why ATK classes are final. I'm trying to use accessibility through Gtk object but it's always a failure. Could you explain me how I should use the library or how can I adapt the generation of ATK to use the library ?

**Comment by afcowie** [comment]
Hello. You're certainly welcome to adapt the library in any way you see fit; There's no one actively maintaining it right now.

When we added coverage for the ATK library, it was very young and _we_ didn't have anyone using it. So I daresay that it could use some significant improvement.

**Comment by giuseppe1993** [comment]
Hello Andrew,
I'd like to use your library; however, since there's no ATK available, I'd deal with completing the Java-Gnome part. I don't know how to edit it since you use several steps to generate the library.
I got the fact that defs files play an important role in the process but the whole thing isn't quite clear to me (and I have to blame my not-so-deep knowledge of Makefile).
If you could tell me in short how to proceed I'd be much grateful; I can leave a private contact info of mine to go on into our conversation.

**Comment by afcowie** [comment]
@giuseppe1993 
I have done a quick branch where I renamed the text files to _.markdown_ so that GitHub will render them properly. See [START](https://github.com/afcowie/java-gnome/blob/filenames/doc/design/START.markdown). These documents were written almost 13 years ago, but they describe the transition from **java-gnome** 2.x to 4.0 and its design fairly well. In particular the _5a-Architecture_ document tells you most of what you need, or rather, what _we_ needed at the time. I hope it helps.

I don't know if the library is still building or what needs to be done to catch up with modern GTK. Good luck!
